src/backend/distributed/utils/reference_table_utils.c

```diff
@@ -418,6 +450,77 @@ DeleteAllReferenceTablePlacementsFromNode(char *workerName, uint32 workerPort)
 
 
 /*
+ * EnsureReferenceTablesReplicatedToAllNodes errors out if it there exists any
+ * under-replicated reference tables in the cluster.
+ */
+void
+EnsureReferenceTablesReplicatedToAllNodes()
+{
+	List *referenceTableOidList = ReferenceTableOidList();
+	ListCell *referenceTableOidCell = NULL;
+	List *workerNodeList = WorkerNodeList();
+	int workerCount = list_length(workerNodeList);
+
+	foreach(referenceTableOidCell, referenceTableOidList)
+	{
+		Oid tableId = lfirst_oid(referenceTableOidCell);
+		DistTableCacheEntry *cacheEntry = DistributedTableCacheEntry(tableId);
+		ShardInterval *shardInterval = cacheEntry->sortedShardIntervalArray[0];
+		uint64 shardId = shardInterval->shardId;
+		List *finalizedShardPlacementList = FinalizedShardPlacementList(shardId);
+
+		if (list_length(finalizedShardPlacementList) != workerCount)
+		{
+			WorkerNode *missingWorkerNode =
+				FindUnderReplicatedWorkerNode(finalizedShardPlacementList);
+
+			ereport(ERROR, (errcode(ERRCODE_OBJECT_NOT_IN_PREREQUISITE_STATE),
+							errmsg("operation is not allowed because at least one of "
```

Comment:
> operation is not allowed because nodename:nodeport is not active due to under-replicated reference table

message:  operation is not allowed because nodename:nodeport is not active
detail: at least one of the reference tables is under-replicated: \"%s\"
hint: keep the same